src/main/java/com/marklogic/mgmt/PayloadParser.java

```diff
@@ -28,20 +28,31 @@ public String getPayloadName(String payload, String idFieldName) {
         return getPayloadFieldValue(payload, idFieldName);
     }
 
-    public String getPayloadFieldValue(String payload, String fieldName) {
+	public String getPayloadFieldValue(String payload, String fieldName) {
+    	return getPayloadFieldValue(payload, fieldName, true);
+	}
+
+	public String getPayloadFieldValue(String payload, String fieldName, boolean throwErrorIfNotFound) {
```

Comment:
Added this overridden method so that if group-name isn't in the payload, no error is thrown.